lib/elixir/lib/module/types.ex

```diff
@@ -349,6 +349,7 @@ defmodule Module.Types do
 
   defp format_traces(traces, simplify?) do
     traces
+    |> Enum.uniq()
```

Comment:
Maps currently accumulate traces as we do multiple passes, this removes the duplicate traces. Not sure if there is a better solution... but I would rather do the extra work when formatting errors than rather as we unify.